opentelemetry-sdk/src/metrics/mod.rs

```diff
@@ -57,3 +57,86 @@ pub use meter_provider::*;
 pub use periodic_reader::*;
 pub use pipeline::Pipeline;
 pub use view::*;
+
+#[cfg(all(test, feature = "testing"))]
+mod tests {
+    use super::*;
+    use crate::{runtime, testing::metrics::InMemoryMetricsExporter};
+    use opentelemetry::{
+        metrics::{MeterProvider as _, Unit},
+        KeyValue,
+    };
+    use tokio::time::{sleep, Duration};
+
+    #[tokio::test]
+    async fn counter_aggregation() {
+        // Run this test with stdout enabled to see output.
+        // cargo test counter --features=metrics,testing -- --nocapture
+
+        // Arrange
+        let exporter = InMemoryMetricsExporter::default();
+        let reader = PeriodicReader::builder(exporter.clone(), runtime::Tokio).build();
+        let meter_provider = SdkMeterProvider::builder().with_reader(reader).build();
+
+        // Act
+        let meter = meter_provider.meter("test");
+        let counter = meter
+            .u64_counter("my_counter")
+            .with_unit(Unit::new("my_unit"))
+            .init();
+        counter.add(1, &[KeyValue::new("key1", "value1")]);
+        counter.add(1, &[KeyValue::new("key1", "value1")]);
+        counter.add(1, &[KeyValue::new("key1", "value1")]);
+        counter.add(1, &[KeyValue::new("key1", "value1")]);
+        counter.add(1, &[KeyValue::new("key1", "value1")]);
+
+        counter.add(1, &[KeyValue::new("key1", "value2")]);
+        counter.add(1, &[KeyValue::new("key1", "value2")]);
+        counter.add(1, &[KeyValue::new("key1", "value2")]);
+
+        // TODO: Unable to get this to work, as
+        // flush is taking forever.
+        // println!("flush started");
+        // meter_provider.force_flush().unwrap();
+        // println!("flush finished");
+        // The sleep is temporary workaround to
+        // unblock progress on writing tests like
+        // this.
+        sleep(Duration::from_millis(100)).await;
```

Comment:
its risky to check-in this as-is, as this won't be reliable. So need to understand why flush/shutdown is not working inside unittests, and fix it..
@jtescher @TommyCpp Either of you can help given you have much more knowledge of tokio runtime/metric sdk internals!

Reply:
Tokio test by default uses single thread runtime, which can be tricky. See [here](https://docs.rs/tokio/latest/tokio/attr.test.html)

Maybe try annotate the test to use multi thread runtime

Reply:
That did fix the Flush-not-progressing issue! Thank you!
I'll try to tweak the tests to make it reliable now.